### ISSUE TITLE
Switch to solid-client-authn-js

### ISSUE THREAD

**Issue by erikhofer** [NoelDeMartin/soukai-solid#6]
The readme shows the use of [`solid-auth-client`](https://github.com/solid/solid-auth-client) which is deprecated. The current default is [`solid-client-authn-js`](solid-client-authn-js). It works fine at runtime, but TypeScript rejects the `fetch` method. You have to do something like this to make it compile.

```ts
import { Session } from '@inrupt/solid-client-authn-browser'
import Soukai from 'soukai'
import SoukaiSolid, { SolidEngine } from 'soukai-solid'

const session = new Session()

SoukaiSolid.loadSolidModels()
// eslint-disable-next-line react-hooks/rules-of-hooks
Soukai.useEngine(new SolidEngine(session.fetch as any))
```

Also the readme should be updated.

**Comment by NoelDeMartin** [comment]
Hey, thanks for opening the issue. Yeah I'm aware of the Typescript issue because I've also faced it using Inrupt's package.

However, I don't think there's a problem with the readme, I already mention that authentication is outside of the scope of this package and I've linked to [Ramen](https://github.com/NoelDeMartin/ramen/) for anyone who wants to learn more.

I also don't agree that `solid-auth-client` is deprecated, because Inrupt's package does not support authenticating with old servers and switching to that library will break existing deployments. I don't know how many people would suffer from this, and I know I'm probably in the minority with this opinion. But personally I'll continue supporting legacy authentication for at least a couple of years. If you're interested to learn more I've been talking about that in this issue: https://github.com/NoelDeMartin/media-kraken/issues/9

Having said that, thanks again for opening the issue! I'll leave it open until I fix the Typescript definitions (which I probably won't do until the next release, because I also want to improve the base library's types, see https://github.com/NoelDeMartin/soukai/issues/1).

And I agree with rewriting the readme as an improvement, but I think I'll just remove any references to `solid-auth-client` and I won't use any authentication library, I'll just leave the link to Ramen. I don't want to use Inrupt's package in the example for a couple of reasons. As I've explained, I don't agree that it's a best practice to only use that one, and also the sample code would be more verbose because you have to instantiate a `Session`, call `handleIncomingRedirect`, etc. And I think it'd be too much boilerplate for something that is outside of this package's scope.

**Comment by erikhofer** [comment]
@NoelDeMartin Okay, I just read the deprecation notice in the readme of `solid-auth-client` and wasn't aware that the situation is more complex. Thanks for linking to more information about this.

I agree that authentication is out of the scope for this library. I think it's okay to completely get rid of it in the readme. Just make sure keep the part about passing a custom `fetch` function to the library.

Also, thanks for putting all the effort into this! :)

**Comment by NoelDeMartin** [comment]
Fixed in [v0.4.1](https://github.com/NoelDeMartin/soukai-solid/releases/tag/v0.4.1).
